src/Azure.Functions.Cli/Common/DurableManager.cs

```diff
@@ -101,19 +106,51 @@ private void SetStorageServiceAndTaskHubClient(ref AzureStorageOrchestrationServ
             }
         }
 
+        private void Initialize(ref AzureStorageOrchestrationService orchestrationService, ref TaskHubClient taskHubClient, string connectionStringKey = null, string taskHubName = null)
+        {
+            CheckAssemblies();
+            SetStorageServiceAndTaskHubClient(ref _orchestrationService, ref _client, connectionStringKey, taskHubName);
+        }
+
+        private void CheckAssemblies()
+        {
+            // Retrieve list of DurableTask.AzureStorage DLL files
+            var assemblyFilePaths = Directory.GetFiles(Environment.CurrentDirectory, $"{DurableAzureStorageExtensionName}", SearchOption.AllDirectories).ToList<string>();
+            
+            if (assemblyFilePaths.Count() == 0)
+            {
+                ColoredConsole.WriteLine(Yellow($"Could not find {DurableAzureStorageExtensionName}. The functions host must be running a" +
+                    $" Durable Functions app in order for Durable Functions CLI commands to work."));
+                return;
+            }
+
+            // Convert them to a sorted list of FileInfo objects. Sort by the last modified time of the enclosing bin folder
+            var assemblyFiles = assemblyFilePaths.Select(assemblyFilePath => new FileInfo(assemblyFilePath)).ToList().OrderBy(file => file.Directory.LastWriteTime);
+
+            var minimumExtensionVersion = new Version(MinimumDurableAzureStorageExtensionVersion);
+
+            var mostRecentAssembly = FileVersionInfo.GetVersionInfo(assemblyFiles.LastOrDefault().FullName);
```

Comment:
You don't want `.LastOrDefault()` because that implies that you expect there could be a `null` value (in which case you would hit a `NullReferenceException`). Instead you should just use `Last()` to make it clear that you expect there to be a list item in the list (and if that's not the case, the error message is more clear).